File: ros_ws/src/my_robot_tutorials/src/led_panel.cpp
#include <ros/ros.h>
#include <my_robot_msgs/SetLedPanel.h>

bool ledPanel[] = {false, false, false};

bool handleServiceSetLed(my_robot_msgs::SetLedPanel::Request &req, my_robot_msgs::SetLedPanel::Response &res)
{
    if(req.led_number > 3 || req.led_number < 1) {
        ROS_INFO("Invalid led number");
        res.success = false;
    }
    else {
        ledPanel[req.led_number - 1] = req.led_state;
        res.success = true;
        ROS_INFO("Led panel: %d, %d, %d", ledPanel[0], ledPanel[1], ledPanel[2]);
    }

    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "led_panel");
    ros::NodeHandle nh;
    ros::ServiceServer service = nh.advertiseService("set_led", handleServiceSetLed);     

    ROS_INFO("led_panel service has been created");

    ros::spin();

    return 0;
}

File: ros_ws/src/my_robot_tutorials/scripts/led_panel.py
#!/usr/bin/env python

import rospy
from my_robot_msgs.srv import SetLedPanel


led_panel = [False, False, False]


def main(args=None):
    rospy.init_node("led_panel")

    service = rospy.Service("set_led", SetLedPanel, handle_set_led_panel)    

    rospy.loginfo("led_panel node has been created")

    rospy.spin()


def handle_set_led_panel(req):
    
    if req.led_number > 3 or req.led_number < 1:
        rospy.loginfo("Incorrect led number")
        return False
    else:
        global led_panel
        led_panel[req.led_number - 1] = req.led_state
        rospy.loginfo("Led panel: " + str(led_panel[0]) + ", " + str(led_panel[1]) + ", " + str(led_panel[2]))
        return True


if __name__ == "__main__":
    main()